src/main/java/org/sonarqube/gradle/SonarPropertyComputer.java

```diff
@@ -143,12 +163,74 @@ private void computeSonarProperties(Project project, Map<String, Object> propert
     for (Project childProject : enabledChildProjects) {
       String moduleId = childProject.getPath();
       moduleIds.add(moduleId);
-      String modulePrefix = (prefix.length() > 0) ? (prefix + "." + moduleId) : moduleId;
-      computeSonarProperties(childProject, properties, modulePrefix);
+      String modulePrefix = !prefix.isEmpty() ? (prefix + "." + moduleId) : moduleId;
+      computeDefaultProperties(childProject, properties, modulePrefix);
     }
+
     properties.put(convertKey("sonar.modules", prefix), String.join(",", moduleIds));
   }
 
+  private boolean shouldApplyScanAll(Project project, Map<String, Object> properties) {
+    // when the parent module is skipped, the properties are empty thus the scan all logic is not applied
+    var scanAllValue = (String) properties.getOrDefault(SONAR_GRADLE_SCAN_ALL, "false");
+    var scanAllEnabled = "true".equalsIgnoreCase(scanAllValue.trim());
+
+    if (scanAllEnabled) {
+      LOGGER.info("Parameter sonar.gradle.scanAll is enabled. The scanner will attempt to collect additional sources.");
+
+      // Collecting the properties configured in the Gradle build configuration
+      var sonarProps = new SonarProperties(new HashMap<>());
+      actionBroadcastMap.get(project.getPath()).execute(sonarProps);
+
+      var sourcesOrTestsAlreadySet = Stream.of(System.getProperties().keySet(), Utils.loadEnvironmentProperties(System.getenv()).keySet(), sonarProps.getProperties().keySet())
+        .flatMap(Collection::stream)
+        .anyMatch(k -> ScanProperties.PROJECT_SOURCE_DIRS.equals(k) || ScanProperties.PROJECT_TEST_DIRS.equals(k));
```

Comment:
let's check if the property ends with rather than equals to have consistency with the [maven implementation](https://github.com/SonarSource/sonar-scanner-maven/blob/5ca170970f60a41068afbae3cf631db0f349a15c/src/main/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapper.java#L169)